data/transactions/logic/eval.go

```diff
@@ -731,6 +741,15 @@ func (cx *evalContext) checkStep() (int, error) {
 	if (cx.runModeFlags & spec.Modes) == 0 {
 		return 0, fmt.Errorf("%s not allowed in current mode", spec.Name)
 	}
+
+	if spec.Name == "log" {
+		cx.logCalls++
+	}
```

Comment:
By putting this in Check(), the program will only be able to have 8 log calls in it.  But we want to enforce is that only 8 log calls actually happen.  So I would not track this here, but rather in Eval().

This also means you can place the increment in opLog, instead of checking the name and incrementing here.